File: java/4. Coeficientes de una ecuación, 2 grado/Coeficientes.java
import java.util.Scanner;
import java.util.InputMismatchException;
import java.util.*;
public class Coeficientes{
    public static void main(String[] args){
	Scanner mEntrada = new Scanner(System.in);
	System.out.println("Programa que factoriza una ecuación cuadrática.");
	System.out.println("Organice la ecuación de la siguiente forma: \n Ax² +Bx + C = 0");
	try{
	    System.out.println("Ingrese el valor de A");
	    float A = mEntrada.nextFloat();
	    if ((int)A == 0){
		System.out.println("A no puede ser 0.");
	    }
	    else {
		System.out.println("Ingrese el valor de B");
		float B = mEntrada.nextFloat();
		
		System.out.println("Ingrese el valor de C");
		float C = mEntrada.nextFloat();
		//		System.out.println("####################################################################################");
		System.out.println("La ecuación queda de la siguiente forma:\n" +  A+ "x² + "+B+"x + "+ C);


		int D = isReal(A,B,C);
		if (D == 0){
		    //		    System.out.println("####################################################################################");
		    System.out.println("Las raices de la ecuación son: ");
		    double x1 = (double) numReal(A,B,C).get(0);
		    double x2 = (double) numReal(A,B,C).get(1);
		    double x11 = (double) 2*A;
		    double X1 = x1/x11;
		    double X2 = x2/x11;
		    System.out.println("x1= " + x1 + "/"+ 2*A+ "\t x2= " + x2+ "/"+ 2*A);
		    //		    System.out.println("####################################################################################");
		    System.out.println("Factorizado queda de la siguiente forma: ");
		    System.out.println(A+"*(x + (" + x1 + "/"+ 2*A+ "))*(x + ("+ x2+ "/"+ 2*A+"))\t o\n" +A+"*(x + (" + X1 + "))*(x + ("+ X2+"))\n ");

		}
		
		if (D==1){
		    System.out.println("####################################################################################");
		    System.out.println("Las raices de la ecuación son: ");
		    System.out.println("x1 = "+ B +"/"+2*A+" \t x2 = "+ B +"/"+2*A);
		    //		    System.out.println("####################################################################################");

		    System.out.println("Factorizado queda de la siguiente forma: ");
		    System.out.println(A+"*(x + ("+B+"/"+2*A+"))²");		    		    System.out.println(A+"*(x + ("+B/(2*A)+"))²");
		}

		if (D == 2){
		    //		    System.out.println("####################################################################################");
		    System.out.println("Las raices de la ecuación compleja son ");
		    double x1 = (double) numComplex(A,B,C).get(0);
		    double x2 = (double) numComplex(A,B,C).get(1);
		    System.out.println("x1= " + x1 +  " + (" + x2 + ")i \t x2= " + x1 +  " - (" +x2 +")i");
		    //		    System.out.println("####################################################################################");
		    System.out.println("Factorizado queda de la siguiente forma: ");
		    System.out.println(A+ "*(x +((" + x1 +  ") + (" + x2 + ")i))*(x + ((" + x1 +  ") - (" +x2 +")i))");
		}

	    }
	    
	}catch(InputMismatchException ime){
	    System.out.println("No introduzca carecteres, se esperan números.");
	}
    }
    	 static int isReal(float a, float b, float c){
	    float D;
	    int R;
	    D = b*b - 4*a*c;
	    R = 0;
	    if (D>0){
	       R = 0;
	    }
	    if (D==0){
		R = 1;
	    }
	    if (D<0){
		R = 2;
	    }
	    return R;
	}
    
     static List numReal(float a, float b, float c){
	 List<Double> Raices = new ArrayList<Double>();
	 double D;
	 double R1;
	 double R2;
	 D = Math.sqrt(b*b - 4*a*c);
	 R1 = (D-b)*(-1);
	 R2 = (-D-b)*(-1);
	 Raices.add(R1);
	 Raices.add(R2);
	 return Raices;	
	
    }

    static List numComplex(float a, float b, float c){
	List<Double> Raices = new ArrayList<Double>();
	double D;
	double R1;
	double R2;
	double B = (double) b;
	double A = (double) a;

	D = Math.sqrt(4*a*c- b*b);
	R1 = B/(2*A);
	R2 = D*(-1)/(2*A);
	Raices.add(R1);
	Raices.add(R2);
	return Raices;	
	
    }
}

File: python/4. Coeficientes de una ecuación, 2 grado/Coeficientes.py
#*-* coding:utf-8*-*
#Diego Alberto Parra Garzón
#Bogotá D.C.,  Colombia;
import math

def isReal(a, b, c):
    d = b*b - 4*a*c
    d = int(d)
    output = 0
    if (d>0):
        print "Es real"
        output = 2
    if (d<0):
        print "Es complejo"
        output = 1
    if(d==0):
        print "Es real básic"
        output = 0
    return output

def numComplex(a,b,c):
    d = math.sqrt(4*a*c - b*b)
    A = b/(2*a)
    B = (d/(2*a))*(-1)
    return A, B

def numReal(a,b,c):
    d = math.sqrt(b*b-4*a*c)
    A = (-b + d)*(-1)
    B = (-b - d)*(-1)
    return A, B

print "Programa que cálcula las raices de una ecuación de grado 2"
print "Organice la ecuación de la siguiente forma: \n Ax² + Bx + C = 0"
A = raw_input("Ingrese el valor de A: ")
A = float(A)
if int(A) == 0:
    print "A no puede ser 0"
    exit()
    quit()
else:
    B = raw_input("Ingrese el valor de B: ")
    B = float(B)
    C = raw_input("Ingrese el valor de C: ")
    C = float(C)

    print "la ecuación queda de la siguiente forma: \n "+str(A)+"x² + "+str(B)+"x + "+str(C)+" = 0"

    
    NumberFamily = isReal(A,B,C)
    # 0 si es real fácil
    # 1 si es complejo
    # 2 si es real
    if (NumberFamily == 0):
        print "Las raices de la ecuaciión son: \n x1 ="+ str(B) + "/"+str(2*A) +" \t x2 = "+ str(B) + "/"+str(2*A)
        print "Realizando las operaciones. "
        print "x1 ="+ str(B/(2*A)) +" \t x2 = "+ str(B/(2*A))
        print "##############################################"
        print "Factorizando queda de la siguiente forma: "
        print str(A)+"*(x + ("+ str(B/(2*A))+"))²"
        
    if (NumberFamily == 1):
        x1, x2 =  numComplex(A,B,C)
        X1 = "(("+str(x1)+")+("+str(x2)+")i)"
        X2 =  "(("+str(x1)+"-)+("+str(x2)+")i)"
        print "Las raices de la ecuación compleja son: \n x1 = " + str(x1)+" + ("+ str(x2)+")i \n x2 = " + str(x1)+" - ("+ str(x2)+")i"
        
        print "##############################################"
        print "Factorizando queda de la siguiente forma: "
        print str(A)+"*(x +"+X1+" )*(x +"+X2+")"
        
    if (NumberFamily == 2):
        x1, x2 =  numReal(A,B,C)
        X1 =  str(x1) + "/"+str(2*A)  
        X2 =   str(x2) + "/"+str(2*A)
        X11 = str(x1/(2*A))
        X22 = str(x2/(2*A))
        print "Las raices de la ecuaciión son: \n x1 ="+X1 +" \t x2 = "+ X2
        print "Realizando las operaciones. "
        print "x1 ="+ X11 +" \t x2 = "+ X22
        print "##############################################"
        print "Factorizando queda de la siguiente forma: "
        print str(A)+"*(x +("+X1+"))*(x +("+X2+"))"

